compiler/src/typed/typemod.re

```diff
@@ -1051,6 +1051,7 @@ let initial_env = () => {
 let type_implementation = (prog: Parsetree.parsed_program) => {
   let sourcefile = prog.prog_loc.loc_start.pos_fname;
   let module_name = prog.module_name.txt;
+  Env.clear_imports();
```

Comment:
Why is this needed now?

Reply:
This basically used to happen as a part of the full compile pipeline, but now that things are moved around, we reset it where it's relevant.